### DISCUSSION TITLE
Non convergence in Tensor Mechanics and Heat Conduction coupling

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#22554] [Q&A General] [answered]
Hi！

I am trying to make a code using Tensor Mechanics Action to calculate the thermal expansion with a temperature transient.  When I do the simulation，the case does not converge. I found something wrong with the mesh as shown the below figure. If I increase the size of the mesh, the accuracy of the case  will not be reached. How can I deal with the problem.

Thanks in advance for your help!
![1](https://user-images.githubusercontent.com/54804946/199227310-f04fafcb-50d2-473f-86fd-c31734054c96.png)

Mao

**Comment by GiudGiud** [comment, marked as answer]
Hello

What are seeing on this figure?
Is this a slice of a 3D mesh? Are these elements?

Guillaume

**Reply by Salma-Mao** [reply]
This is a slice of a 3D mesh. The shape of the mesh is rectangle.   When I do the simulation, the shape of mesh is changed. The case cannot converge.

**Reply by GiudGiud** [reply]
what are the curved surfaces showing? are they cuts of elements? why are they so thin? Do they start out like this?

Deformed mesh issues are often solved by reducing the time step.
Could you please attach the (non)convergence simulation log?

**Reply by Salma-Mao** [reply]
_**what are the curved surfaces showing? are they cuts of elements? why are they so thin? Do they start out like this?**_

The mesh is shown as below at the beginning.  I cut elements  in the paraview. Then, I found the shape of the mesh had been changed.


![image](https://user-images.githubusercontent.com/54804946/200721433-e065e051-50d8-40ca-bb2e-665294ce2d6a.png)


_**Deformed mesh issues are often solved by reducing the time step.
Could you please attach the (non)convergence simulation log?**_

I reduce the time step, but the case still does not converge. 

the (non)convergence simulation log

Time Step 8129, time = 0.00162408, dt = 1.95312e-10
 Solve Converged!
  Finished Solving                                                                       [ 10.81 s] [   28 MB]

Time Step 8130, time = 0.00162408, dt = 3.90625e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [166.16 s] [  -15 MB]
Aborting as solve did not converge

Time Step 8130, time = 0.00162408, dt = 1.95312e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [165.35 s] [  -13 MB]
Aborting as solve did not converge

Time Step 8130, time = 0.00162408, dt = 1e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50


**Time Step 8129, time = 0.00162408, dt = 1.95312e-10
 Solve Converged!
  Finished Solving                                                                       [ 10.81 s] [   28 MB]

Time Step 8130, time = 0.00162408, dt = 3.90625e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [166.16 s] [  -15 MB]
Aborting as solve did not converge

Time Step 8130, time = 0.00162408, dt = 1.95312e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [165.35 s] [  -13 MB]
Aborting as solve did not converge

Time Step 8130, time = 0.00162408, dt = 1e-10
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50


*** ERROR ***
The following error occurred in the object "TimeStepper", of type "ConstantDT".

Solve failed and timestep already at or below dtmin, cannot continue!

*** ERROR ***
The following error occurred in the object "TimeStepper", of type "ConstantDT".

Solve failed and timestep already at or below dtmin, cannot continue!

**Reply by GiudGiud** [reply]
can you print the non linear residual in Outputs

also is 1e-10 appropriate? A time step that is too small can also be an issue

**Reply by Salma-Mao** [reply]
The time step is 1e-7. The non linear residual in Outputs can not be printed. Because the interval is 40 in Outputs.

**Reply by GiudGiud** [reply]
ok let s have a timestep slightly close to 1e-7 then.
The interval in Outputs can be removed so we can see the log

**Reply by Salma-Mao** [reply]
Hello, the log
Time Step 860, time = 0.00017175, dt = 5e-08
 0 Nonlinear |R| = 7.093705e-01
      0 Linear |R| = 7.093705e-01
      1 Linear |R| = 2.733138e-02
 1 Nonlinear |R| = 2.767147e-02
      0 Linear |R| = 2.767147e-02
      1 Linear |R| = 2.010860e-03
 2 Nonlinear |R| = 4.485534e-03
      0 Linear |R| = 4.485534e-03
      1 Linear |R| = 3.930144e-04
      2 Linear |R| = 4.065825e-05
 3 Nonlinear |R| = 3.846285e-03
      0 Linear |R| = 3.846285e-03
      1 Linear |R| = 3.266360e-04
 4 Nonlinear |R| = 3.956858e-03
      0 Linear |R| = 3.956858e-03
      1 Linear |R| = 3.038938e-04
 5 Nonlinear |R| = 3.806141e-03
      0 Linear |R| = 3.806141e-03
      1 Linear |R| = 2.947303e-04
 6 Nonlinear |R| = 3.935276e-03
      0 Linear |R| = 3.935276e-03
      1 Linear |R| = 3.073631e-04
 7 Nonlinear |R| = 3.807751e-03
      0 Linear |R| = 3.807751e-03
      1 Linear |R| = 2.939515e-04
 8 Nonlinear |R| = 3.935097e-03
      0 Linear |R| = 3.935097e-03
      1 Linear |R| = 3.075556e-04
 9 Nonlinear |R| = 3.807777e-03
      0 Linear |R| = 3.807777e-03
      1 Linear |R| = 2.938990e-04
10 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075705e-04
11 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938939e-04
12 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075706e-04
13 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938941e-04
14 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075712e-04
15 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938940e-04
16 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075715e-04
17 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938937e-04
18 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
19 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
20 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
21 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
22 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
23 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
24 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
25 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
26 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
27 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
28 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
29 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
30 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
31 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
32 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
33 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
34 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
35 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
36 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
37 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
38 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
39 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
40 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
41 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
42 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
43 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
44 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
45 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
46 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
47 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
48 Nonlinear |R| = 3.935089e-03
      0 Linear |R| = 3.935089e-03
      1 Linear |R| = 3.075716e-04
49 Nonlinear |R| = 3.807779e-03
      0 Linear |R| = 3.807779e-03
      1 Linear |R| = 2.938938e-04
50 Nonlinear |R| = 3.935089e-03
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [324.84 s] [   20 MB]
Aborting as solve did not converge

Time Step 860, time = 0.000171725, dt = 2.5e-08
 0 Nonlinear |R| = 7.091746e-01
      0 Linear |R| = 7.091746e-01
      1 Linear |R| = 2.408680e-02
 1 Nonlinear |R| = 2.446325e-02
      0 Linear |R| = 2.446325e-02
      1 Linear |R| = 1.940166e-03
 2 Nonlinear |R| = 4.499643e-03
      0 Linear |R| = 4.499643e-03
      1 Linear |R| = 3.886310e-04
      2 Linear |R| = 4.095673e-05
 3 Nonlinear |R| = 3.845086e-03
      0 Linear |R| = 3.845086e-03
      1 Linear |R| = 3.295423e-04
 4 Nonlinear |R| = 3.959255e-03
      0 Linear |R| = 3.959255e-03
      1 Linear |R| = 3.059037e-04
 5 Nonlinear |R| = 3.804623e-03
      0 Linear |R| = 3.804623e-03
      1 Linear |R| = 2.970830e-04
 6 Nonlinear |R| = 3.937440e-03
      0 Linear |R| = 3.937440e-03
      1 Linear |R| = 3.094788e-04
 7 Nonlinear |R| = 3.806273e-03
      0 Linear |R| = 3.806273e-03
      1 Linear |R| = 2.962768e-04
 8 Nonlinear |R| = 3.937257e-03
      0 Linear |R| = 3.937257e-03
      1 Linear |R| = 3.096787e-04
 9 Nonlinear |R| = 3.806300e-03
      0 Linear |R| = 3.806300e-03
      1 Linear |R| = 2.962212e-04
10 Nonlinear |R| = 3.937249e-03
      0 Linear |R| = 3.937249e-03
      1 Linear |R| = 3.096943e-04
11 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962162e-04
12 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096957e-04
13 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962162e-04
14 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096960e-04
15 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962151e-04
16 Nonlinear |R| = 3.937249e-03
      0 Linear |R| = 3.937249e-03
      1 Linear |R| = 3.096962e-04
17 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
18 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096964e-04
19 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
20 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
21 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
22 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
23 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
24 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
25 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
26 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
27 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
28 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
29 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
30 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
31 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
32 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
33 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
34 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
35 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
36 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
37 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
38 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
39 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
40 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096964e-04
41 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
42 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
43 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
44 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096964e-04
45 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
46 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096965e-04
47 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
48 Nonlinear |R| = 3.937248e-03
      0 Linear |R| = 3.937248e-03
      1 Linear |R| = 3.096964e-04
49 Nonlinear |R| = 3.806302e-03
      0 Linear |R| = 3.806302e-03
      1 Linear |R| = 2.962154e-04
50 Nonlinear |R| = 3.937248e-03
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [324.88 s] [  -20 MB]
Aborting as solve did not converge

**Reply by GiudGiud** [reply]
What are the executioner parameters? The linear residuals don’t decrease enough

**Reply by Salma-Mao** [reply]
Hello， The executioner is below.
```
[Executioner]
  type = Transient
  solve_type = PJFNK
  petsc_options = '-snes_ksp_ew'
  nl_rel_tol = 1e-3
  nl_abs_tol = 1e-3

  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre  boomeramg '

   line_search = 'none'

  l_max_its = 100

  end_time = 6.9e-3
  dt = 1e-7
  dtmin = 1e-10
  automatic_scaling = true
[]
```

**Reply by GiudGiud** [reply]
the non linear convergence criteria are very coarse. Especially with automatic scaling active. Can you tighten that a lot? Maybe 1e-8 instead of 1e-3

**Reply by Salma-Mao** [reply]
I replace 1e-3 with 1e-8. The case still not convergence. In the case, I find that the convergence of the calculation is a little poor.

Time Step 1040, time = 0.0002079, dt = 2e-07
 0 Nonlinear |R| = 5.670698e-01
      0 Linear |R| = 5.670698e-01
      1 Linear |R| = 5.656334e-02
 1 Nonlinear |R| = 5.652462e-02
      0 Linear |R| = 5.652462e-02
      1 Linear |R| = 2.936858e-03
 2 Nonlinear |R| = 2.936574e-03
      0 Linear |R| = 2.936574e-03
      1 Linear |R| = 2.908964e-04
      2 Linear |R| = 2.798045e-05
      3 Linear |R| = 2.114183e-06
 3 Nonlinear |R| = 2.116220e-06
      0 Linear |R| = 2.116220e-06
      1 Linear |R| = 2.644098e-07
      2 Linear |R| = 3.705461e-08
      3 Linear |R| = 2.373776e-08
      4 Linear |R| = 1.782929e-08
      5 Linear |R| = 1.233029e-08
      6 Linear |R| = 9.290520e-09
      7 Linear |R| = 6.581831e-09
      8 Linear |R| = 5.445464e-09
      9 Linear |R| = 4.664535e-09
     10 Linear |R| = 4.228118e-09
     11 Linear |R| = 3.722396e-09
     12 Linear |R| = 3.270257e-09
     13 Linear |R| = 2.572205e-09
     14 Linear |R| = 1.858021e-09
     15 Linear |R| = 1.358547e-09
     16 Linear |R| = 8.984392e-10
     17 Linear |R| = 5.854350e-10
     18 Linear |R| = 3.701566e-10
     19 Linear |R| = 2.496561e-10
     20 Linear |R| = 1.608496e-10
     21 Linear |R| = 1.102003e-10
     22 Linear |R| = 7.927184e-11
     23 Linear |R| = 6.380655e-11
     24 Linear |R| = 5.195455e-11
     25 Linear |R| = 4.228379e-11
     26 Linear |R| = 3.646806e-11
     27 Linear |R| = 3.341420e-11
     28 Linear |R| = 3.022875e-11
     29 Linear |R| = 2.741380e-11
     **30 Linear |R| = 2.745361e-10**
     31 Linear |R| = 2.439306e-11
     32 Linear |R| = 2.196101e-11
     33 Linear |R| = 1.970708e-11
     34 Linear |R| = 1.699910e-11
 4 Nonlinear |R| = 2.511046e-10
 Solve Converged!
  Finished Solving

**Reply by GiudGiud** [reply]
The linear solve is still calling it converged too early. What do the parameters look like now?

**Reply by Salma-Mao** [reply]
There is a strange thing. The case converges at a certain temperature. When I change the temperature, the  case doesn't converge.  
Time Step 10800, time = 0.00431555, dt = 1e-07
 0 Nonlinear |R| = 7.247351e-01
      0 Linear |R| = 7.247351e-01
      1 Linear |R| = 3.811024e-02
 1 Nonlinear |R| = 3.960667e-02
      0 Linear |R| = 3.960667e-02
      1 Linear |R| = 2.010666e-03
 2 Nonlinear |R| = 1.176742e-02
      0 Linear |R| = 1.176742e-02
      1 Linear |R| = 6.200048e-04
 3 Nonlinear |R| = 1.127051e-02
      0 Linear |R| = 1.127051e-02
      1 Linear |R| = 5.585384e-04
 4 Nonlinear |R| = 1.153546e-02
      0 Linear |R| = 1.153546e-02
      1 Linear |R| = 5.413609e-04
 5 Nonlinear |R| = 1.124045e-02
      0 Linear |R| = 1.124045e-02
      1 Linear |R| = 5.367732e-04
 6 Nonlinear |R| = 1.153239e-02
      0 Linear |R| = 1.153239e-02
      1 Linear |R| = 5.442985e-04
 7 Nonlinear |R| = 1.124036e-02
      0 Linear |R| = 1.124036e-02
      1 Linear |R| = 5.349234e-04
 8 Nonlinear |R| = 1.153230e-02
      0 Linear |R| = 1.153230e-02
      1 Linear |R| = 5.449348e-04
 9 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.346405e-04
10 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450494e-04
11 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345917e-04
12 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450705e-04
13 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345830e-04
14 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450736e-04
15 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345807e-04
16 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450742e-04
17 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345809e-04
18 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450745e-04
19 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345806e-04
20 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450745e-04
21 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
22 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
23 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
24 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
25 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
26 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
27 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
28 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
29 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
30 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
31 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
32 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
33 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
34 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
35 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
36 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
37 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
38 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
39 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
40 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
41 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
42 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
43 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
44 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
45 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
46 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
47 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
48 Nonlinear |R| = 1.153229e-02
      0 Linear |R| = 1.153229e-02
      1 Linear |R| = 5.450743e-04
49 Nonlinear |R| = 1.124037e-02
      0 Linear |R| = 1.124037e-02
      1 Linear |R| = 5.345805e-04
50 Nonlinear |R| = 1.153229e-02
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 50
 Solve Did NOT Converge!
  Finished Solving                                                                       [ 71.23 s] [    0 MB]
Aborting as solve did not converge

**Reply by GiudGiud** [reply]
Hello

Please paste the executioner block again here.

This convergence history is consistent with really loose convergence criteria

Guillaume

**Reply by Salma-Mao** [reply]
Hello


[Executioner]
  type = Transient
  solve_type = PJFNK
  petsc_options = '-snes_ksp_ew'
  nl_rel_tol = 1e-5
  nl_abs_tol = 1e-5

  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre  boomeramg '


  line_search = 'none'

  l_max_its = 100

  end_time = 9.3e-3
  dt = 4.0e-7
  dtmin = 1e-10
  automatic_scaling = true
[]

**Reply by GiudGiud** [reply]
let's tighten these non linear residuals further please

**Reply by GiudGiud** [reply]
any luck with tightening the convergence criteria?
